servers/mcp-neo4j-memory/src/mcp_neo4j_memory/server.py

```diff
@@ -14,456 +14,556 @@
+  entities: List[Entity]
+  relations: List[Relation]
+
 
 class ObservationAddition(BaseModel):
-    entityName: str
-    contents: List[str]
+  entityName: str
+  contents: List[str]
+
 
 class ObservationDeletion(BaseModel):
-    entityName: str
-    observations: List[str]
+  entityName: str
+  observations: List[str]
+
 
 class Neo4jMemory:
-    def __init__(self, neo4j_driver):
-        self.neo4j_driver = neo4j_driver
-        self.create_fulltext_index()
-
-    def create_fulltext_index(self):
-        try:
-            # TODO , 
-            query = """
-            CREATE FULLTEXT INDEX search IF NOT EXISTS FOR (m:Memory) ON EACH [m.name, m.type, m.observations];
-            """
-            self.neo4j_driver.execute_query(query)
-            logger.info("Created fulltext search index")
-        except neo4j.exceptions.ClientError as e:
-            if "An index with this name already exists" in str(e):
-                logger.info("Fulltext search index already exists")
-            else:
-                raise e
-
-    async def load_graph(self, filter_query="*"):
-        query = """
+  def __init__(self, neo4j_driver):
+    self.neo4j_driver = neo4j_driver
+    self.create_fulltext_index()
+
+  def create_fulltext_index(self):
+    """Create fulltext indexes for nodes and relationships in the Neo4j graph."""
+    index_config = {
+      "indexConfig": {
+        "fulltext.analyzer": "cjk",
+        "fulltext.eventually_consistent": True,  # Use async indexing
+      }
+    }
+
+    # Node index: name, type, and observations
+    node_query = """
+      CREATE FULLTEXT INDEX node_search IF NOT EXISTS
+      FOR (n:Memory)
+      ON EACH [n.name, n.type, n.observations]
+      OPTIONS $config
+      """
+
+    try:
+      self.neo4j_driver.execute_query(node_query, {"config": index_config})
+      logger.info("Full text indexes created successfully.")
+    except neo4j.exceptions.ClientError as e:
+      logger.error(f"Indexing failed with: {str(e)}")
+      raise
+
+  async def check_index_status(self):
+      query = """
+      CALL db.index.fulltext.list()
+      YIELD name, state, populationPercent, type
+      WHERE name IN ['node_search', 'rel_search']
+      RETURN name, state, populationPercent, type
+      ORDER BY name
+      """
+
+      result = self.neo4j_driver.execute_query(query)
+      status = {}
+
+      for record in result.records:
+          status[record["name"]] = {
+              "state": record["state"],
+              "progress": f"{record['populationPercent']}%",
+              "type": record["type"]
+          }
+
+      return {
+          "node_index": status.get("node_search", "not found"),
+      }
+
+  async def fuzzy_search(
+      self,
+      search_term: str,
+      target_type: Optional[str] = None,  # 'node'/'relationship'/None
+      limit: int = 50
+  ) -> Dict[str, List]:
+      """
+      Fuzzy search for nodes and relationships in the knowledge graph using fulltext indexing.
```

Comment:
too much detail, the casual user just wants to find some entities, they will not bother to provide correct type names (if they remember at all).
if you need this level of control as a user you would  probably use the cypher server against the memory datbase.

also not fixing readgraph!